src/views/StandaloneOfficeAddressFiling.vue

```diff
@@ -365,95 +393,93 @@ export default {
+          }
+        } else if (filing.header.bcolAccountNumber) {
+          this.staffPaymentData = {
+            option: StaffPaymentOptions.BCOL,
+            bcolAccountNumber: filing.header.bcolAccountNumber,
+            datNumber: filing.header.datNumber,
+            folioNumber: filing.header.folioNumber,
+            isPriority: filing.header.priority
+          }
+        } else if (filing.header.waiveFees) {
+          this.staffPaymentData = {
+            option: StaffPaymentOptions.NO_FEE
+          }
+        } else {
+          this.staffPaymentData = {
+            option: StaffPaymentOptions.NONE
+          }
+        }
 
-            // load Annual Report fields
-            if (!filing.changeOfAddress) throw new Error('Missing change of address')
-
-            const changeOfAddress = filing.changeOfAddress.offices
-            if (changeOfAddress) {
-              if (changeOfAddress.recordsOffice) {
-                this.addresses = {
-                  registeredOffice: {
-                    deliveryAddress: changeOfAddress.registeredOffice.deliveryAddress,
-                    mailingAddress: changeOfAddress.registeredOffice.mailingAddress
-                  },
-                  recordsOffice: {
-                    deliveryAddress: changeOfAddress.recordsOffice.deliveryAddress,
-                    mailingAddress: changeOfAddress.recordsOffice.mailingAddress
-                  }
-                }
-                // use existing Priority and Waive Fees flags
-                // apply this flag to BCOMPs only
-                this.updateFilingData('add', FilingCodes.ADDRESS_CHANGE_BC, this.staffPaymentData.isPriority,
-                  (this.staffPaymentData.option === StaffPaymentOptions.NO_FEE))
-              } else {
-                this.addresses = {
-                  registeredOffice: {
-                    deliveryAddress: changeOfAddress.registeredOffice.deliveryAddress,
-                    mailingAddress: changeOfAddress.registeredOffice.mailingAddress
-                  }
-                }
-                // use existing Priority and Waive Fees flags
-                // apply this flag to all OTHER entity types
-                this.updateFilingData('add', FilingCodes.ADDRESS_CHANGE_OT, this.staffPaymentData.isPriority,
-                  (this.staffPaymentData.option === StaffPaymentOptions.NO_FEE))
+        // restore COA Date
+        if (filing.header.effectiveDate) {
+          this.coaDate = this.convertUTCTimeToLocalTime(filing.header.effectiveDate).slice(0, 10)
+        } else if (filing.header.date) {
+          this.coaDate = this.convertUTCTimeToLocalTime(filing.header.date).slice(0, 10)
+        } else {
+          throw new Error('Missing effective date')
+        }
+
+        const changeOfAddress = filing.changeOfAddress?.offices
+        if (changeOfAddress) {
+          if (changeOfAddress.recordsOffice && changeOfAddress.registeredOffice) {
```

Comment:
FYI, I'm a little uneasy about checking the data for what it contains -- instead of requiring specific data to exist (ie, for BCOMP). But I didn't change this here. I think we will have to look at this again when we have other entity types with 1 or 2 sets of addresses.

Reply:
In other words, this code checks to see what data is available and then loads it. Instead, I think this code should know what data it expects and then verify that it gets it (else error).